x509/src/pkix_oids.rs

```diff
@@ -188,3 +189,278 @@ pub const PKIX_OCSP_NOCHECK: ObjectIdentifier = ObjectIdentifier::new("1.3.6.1.5
+
+// -------------------------------------------------------------------------------------------------
+// OIDs from PKIX1-PSS-OAEP-Algorithms-2009
+// -------------------------------------------------------------------------------------------------
+//    pkcs-1  OBJECT IDENTIFIER  ::=
+//        { iso(1) member-body(2) us(840) rsadsi(113549) pkcs(1) 1 }
+
+/// sha224WithRSAEncryption  OBJECT IDENTIFIER  ::=  { pkcs-1 14 }
+pub const PKIXALG_SHA224_WITH_RSA_ENCRYPTION: ObjectIdentifier =
+    ObjectIdentifier::new("1.2.840.113549.1.1.14");
+
+/// sha256WithRSAEncryption  OBJECT IDENTIFIER  ::=  { pkcs-1 11 }
+pub const PKIXALG_SHA256_WITH_RSA_ENCRYPTION: ObjectIdentifier =
+    ObjectIdentifier::new("1.2.840.113549.1.1.11");
+
+/// sha384WithRSAEncryption  OBJECT IDENTIFIER  ::=  { pkcs-1 12 }
+pub const PKIXALG_SHA384_WITH_RSA_ENCRYPTION: ObjectIdentifier =
+    ObjectIdentifier::new("1.2.840.113549.1.1.12");
+
+/// sha512WithRSAEncryption  OBJECT IDENTIFIER  ::=  { pkcs-1 13 }
+pub const PKIXALG_SHA512_WITH_RSA_ENCRYPTION: ObjectIdentifier =
+    ObjectIdentifier::new("1.2.840.113549.1.1.13");
+
+/// id-RSAES-OAEP  OBJECT IDENTIFIER  ::=  { pkcs-1 7 }
+pub const PKIXALG_RSAES_OAEP: ObjectIdentifier = ObjectIdentifier::new("1.2.840.113549.1.1.7");
+
+/// id-pSpecified  OBJECT IDENTIFIER  ::=  { pkcs-1 9 }
+pub const PKIXALG_PSPECIFIED: ObjectIdentifier = ObjectIdentifier::new("1.2.840.113549.1.1.9");
+
+/// id-mgf1  OBJECT IDENTIFIER  ::=  { pkcs-1 8 }
+pub const PKIXALG_MGF1: ObjectIdentifier = ObjectIdentifier::new("1.2.840.113549.1.1.8");
+
+/// id-RSASSA-PSS  OBJECT IDENTIFIER  ::=  { pkcs-1 10 }
+pub const PKIXALG_RSASSA_PSS: ObjectIdentifier = ObjectIdentifier::new("1.2.840.113549.1.1.10");
+
+///       sha-1 OBJECT IDENTIFIER ::= { iso(1) identified-organization(3)
+///           oiw(14) secsig(3) algorithm(2) 26 }
+pub const PKIXALG_SHA1: ObjectIdentifier = ObjectIdentifier::new("1.3.14.3.2.26");
+
+/// id-sha224  OBJECT IDENTIFIER  ::=
+///     { joint-iso-itu-t(2) country(16) us(840) organization(1) gov(101)
+///     csor(3) algorithms(4) hashalgs(2) 4 }
+pub const PKIXALG_SHA224: ObjectIdentifier = ObjectIdentifier::new("2.16.840.1.101.3.4.2.4");
+
+/// id-sha256  OBJECT IDENTIFIER  ::=
+///        { joint-iso-itu-t(2) country(16) us(840) organization(1) gov(101)
+///        csor(3) algorithms(4) hashalgs(2) 1 }
+pub const PKIXALG_SHA256: ObjectIdentifier = ObjectIdentifier::new("2.16.840.1.101.3.4.2.1");
+
+/// id-sha384  OBJECT IDENTIFIER  ::=
+///        { joint-iso-itu-t(2) country(16) us(840) organization(1) gov(101)
+///        csor(3) algorithms(4) hashalgs(2) 2 }
+pub const PKIXALG_SHA384: ObjectIdentifier = ObjectIdentifier::new("2.16.840.1.101.3.4.2.2");
+
+/// id-sha512  OBJECT IDENTIFIER  ::=
+///        { joint-iso-itu-t(2) country(16) us(840) organization(1) gov(101)
+///        csor(3) algorithms(4) hashalgs(2) 3 }
+pub const PKIXALG_SHA512: ObjectIdentifier = ObjectIdentifier::new("2.16.840.1.101.3.4.2.3");
```

Comment:
I really don't like the "big pile of OID constants" approach. There has to be a way to organize these better.

Reply:
The OIDs are grouped by module with names derived by capitalization, insertion of underscores and addition of a prefix. I'm OK with this approach, but if there is something desired please let me know.